layer1/wallet/src/wallet_sc.mligo

```diff
@@ -75,13 +78,14 @@ let send ({owner_address; mint_address; bridge_address; ticket_storage} : wallet
 (*  
 Rsponsibility: dispatch based on action to the action handler functions
  *)
-let main (parameter, storage : wallet_parameter * wallet_storage) : wallet_return =
-  if Tezos.source = storage.owner_address then
-      match parameter with
-        | Mint_xtz -> mint_xtz storage 
-        | Mint_xtz_cb ticket -> mint_xtz_cb (ticket,storage)
-        | Send -> send storage
-        | Redeem_xtz -> redeem_xtz storage
-        | Redeem_xtz_cb -> redeem_xtz_cb storage
+let main (parameter, {owner_address; mint_address; bridge_address; ticket_storage} : wallet_parameter * wallet_storage) : wallet_return =
+  let storage = {owner_address = owner_address; mint_address = mint_address ; bridge_address = bridge_address ; ticket_storage = ticket_storage} in 
```

Comment:
I'm just wondering that is it good practice to pattern match variables rather than like `storage.owner_address` ?